jschecker.jar uses a deprecated API

I believe that this warning was introduced with the compiler v20160619. I'll look more into this later but I wanted to create an issue so that it does not get lost.

```
INFO: From Building external/io_bazel_rules_closure/java/com/google/javascript/jscomp/jschecker.jar (8 source files):
Note: external/io_bazel_rules_closure/java/com/google/javascript/jscomp/JsChecker.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
```

Not sure if this is even worth "fixing" or not, but I did a little additional research here.

```
java/com/google/javascript/jscomp/JsChecker.java:221: warning: [deprecation] setIdeMode(boolean) in CompilerOptions has been deprecated
    options.setIdeMode(true);
           ^
1 warning
```

From [CompilerOptions.java](https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/jscomp/CompilerOptions.java#L1895):

```
  /**                                                                           
   * Configures the compiler for use as an IDE backend.  In this mode:          
   * <ul>                                                                       
   *  <li>No optimization passes will run.</li>                                 
   *  <li>The last time custom passes are invoked is                            
   *      {@link CustomPassExecutionTime#BEFORE_OPTIMIZATIONS}</li>             
   *  <li>The compiler will always try to process all inputs fully, even        
   *      if it encounters errors.</li>                                         
   *  <li>The compiler may record more information than is strictly             
   *      needed for codegen.</li>                                              
   * </ul>                                                                      
   *                                                                            
   * @deprecated Some "IDE" clients will need some of these options but not     
   * others. Consider calling setChecksOnly, setAllowRecompilation, etc,        
   * explicitly, instead of calling this method which does a variety of         
   * different things.                                                          
   */
  @Deprecated
  public void setIdeMode(boolean ideMode) {
```

--- a/java/com/google/javascript/jscomp/JsChecker.java
+++ b/java/com/google/javascript/jscomp/JsChecker.java
@@ -27,6 +27,7 @@ import com.google.devtools.build.lib.worker.WorkerProtocol.WorkRequest;
 import com.google.devtools.build.lib.worker.WorkerProtocol.WorkResponse;
 import com.google.javascript.jscomp.CompilerOptions.LanguageMode;
 import com.google.javascript.jscomp.lint.CheckJSDocStyle;
+import com.google.javascript.jscomp.parsing.Config;
 
 import org.kohsuke.args4j.CmdLineException;
 import org.kohsuke.args4j.CmdLineParser;
@@ -218,7 +219,11 @@ public final class JsChecker {
     options.setLanguage(language);
     options.setCodingConvention(convention.convention);
     options.setSkipTranspilationAndCrash(true);
-    options.setIdeMode(true);
+    options.setChecksOnly(true);
+    options.setContinueAfterErrors(true);
+    options.setAllowHotswapReplaceScript(true);
+    options.setPreserveDetailedSourceInfo(true);
+    options.setParseJsDocDocumentation(Config.JsDocParsing.INCLUDE_DESCRIPTIONS_NO_WHITESPACE);
     JsCheckerErrorFormatter errorFormatter = new JsCheckerErrorFormatter(state, compiler);
     errorFormatter.setColorize(true);
     JsCheckerErrorManager errorManager = new JsCheckerErrorManager(errorFormatter);

